Source/SwiftLintBuiltInRules/Rules/Lint/MultilineCallArgumentsRule.swift

```diff
@@ -0,0 +1,150 @@
+import SwiftLintCore
+import SwiftSyntax
+
+@SwiftSyntaxRule(optIn: true)
+struct MultilineCallArgumentsRule: Rule {
+    var configuration = MultilineCallArgumentsConfiguration()
+
+    static let description = RuleDescription(
+        identifier: "multiline_call_arguments",
+        name: "Multiline Call Arguments",
+        description: """
+            Arguments in function and method calls should be either on the same line \
+            or one per line when the call spans multiple lines
+            """,
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("""
+                    foo(
+                    param1: "param1",
+                        param2: false,
+                        param3: []
+                    )
+                    """, configuration: ["max_number_of_single_line_parameters": 2]),
+            Example("""
+                    foo(param1: 1,
+                        param2: false,
+                        param3: [])
+                    """, configuration: ["max_number_of_single_line_parameters": 1]),
+            Example("foo(param1: 1, param2: false)",
+                    configuration: ["max_number_of_single_line_parameters": 2]),
+            Example("Enum.foo(param1: 1, param2: false)",
+                    configuration: ["max_number_of_single_line_parameters": 2]),
+            Example("""
+                    foo(param1: 1)
+                    """,
```

Comment:
One-liners can be written like this:
```suggestion
            Example("foo(param1: 1)",
```